src/core/authentication/authentication.service.ts

```diff
@@ -1,20 +1,44 @@
+import { AxiosError } from 'axios';
+import { HttpService } from '@nestjs/axios';
 import { ConfigService } from '@nestjs/config';
 import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { Session } from '@ory/kratos-client';
 import { ConfigurationTypes, LogContext } from '@common/enums';
 import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
 import { UserService } from '@domain/community/user/user.service';
 import { AgentInfo } from './agent-info';
 import { OryDefaultIdentitySchema } from './ory.default.identity.schema';
 import { NotSupportedException } from '@common/exceptions';
 import { AgentService } from '@domain/agent/agent/agent.service';
+import { getBearerToken } from './get.bearer.token';
+
 @Injectable()
 export class AuthenticationService {
+  private readonly kratosPublicUrl: string;
+  private readonly kratosPrivateUrl: string;
+  private readonly adminPasswordIdentifier: string;
+  private readonly adminPassword: string;
+  private readonly extendAfter: number;
+
   constructor(
     private configService: ConfigService,
     private agentService: AgentService,
     private userService: UserService,
+    private httpService: HttpService,
     @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
-  ) {}
+  ) {
+    this.kratosPublicUrl = this.configService.get(
+      ConfigurationTypes.IDENTITY
+    ).authentication.providers.ory.kratos_public_base_url_server;
+    this.extendAfter = Number(
+      this.configService.get(ConfigurationTypes.IDENTITY).authentication
+        .providers.ory.earliest_possible_extend
+    );
+    // todo: read from env
+    this.kratosPrivateUrl = '';
```

Comment:
These names are a bit confusing, as we have the following endpoints:
  - kratos admin api endpoint --> accessible only inside the cluster (in prod environments)
  - kratos public api endpoint with public URL --> accessible from outside of the cluster
  - kratos public api endpoint with cluster URL --> accessible from inside the cluster only (otherwise we'll be routing requests through traefik)
So my suggestion is:
  - `kratosPrivateUrl` --> `kratosAdminUrlServer` (meaning kratos admin URL from the server container / service)
  - `kratosPublicUrl` --> `kratosPublicUrlServer` (meaning kratos public URL from the server container / service)

Also, I see you are already using `kratos_public_base_url_server` --> at the moment all our configmaps  are misconfigured in all environments to use the public URL of the server there - if we don't fix that, the wrong URL will be taken and this fix won't work on demo / prod (where the admin endpoint is not exposed).

You will also need to add a config entry `kratos_admin_base_url_server` and take `kratosAdminUrlServer` from it.